### PR TITLE
List of statues[WIP]

### PR DESCRIPTION
**Pull request by Gina-Frankel**
This relates to the user story: 
```
AS A site visitor
 I WANT to see a list of statues ,
SO THAT I can understand the context of public artworks
```

Two components have been created

- StatueListContainer - Contains the StatueList and renders a heading 

- StatueList - Iterates over the statue_items and then renders them as **List elements**

The StatueList is going to  be split into a StatueItem component to create greater separation. At the moment the iteration over the statue list is implemented in the render but will be changed. 

The List of Slaves are stored inside a JS object as a database is not necessary at this time. 

CircleCI has been added for **CI** 

If the reviewer could provide feedback on naming, tests, commit messages and anything to do with the how props are being sent that would be greatly appreciated.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-
+      - checkout
+      - node/with-cache:
+          steps:
+            - run: npm install
+            - run: npm test
+workflows:
+    build-and-test:
+      jobs:
+        - build-and-test
```
Comment (benfurber) [inline .circleci/config.yml:17]: Is a line break missing here?

File: public/manifest.json [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-      "src": "logo512.png",
-      "type": "image/png",
-      "sizes": "512x512"
-    }
-  ],
-  "start_url": ".",
-  "display": "standalone",
-  "theme_color": "#000000",
-  "background_color": "#ffffff"
-}
```
Comment (benfurber) [inline public/manifest.json:25]: Why delete this file?

File: public/robots.txt [hunk 3]
```diff
@@ -1,3 +0,0 @@
-# https://www.robotstxt.org/robotstxt.html
-User-agent: *
-Disallow:
```
Comment (benfurber) [inline public/robots.txt:3]: Why delete this file?

File: src/Statues/slavers.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const SlaverStatues = [
```
Comment (benfurber) [inline src/Statues/slavers.js:1]: Can you remove the capital from the folder name please. Keep lower case.

File: src/Statues/slavers.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const SlaverStatues = [ 
+  {
+    statueName: "Edward Colston"
+  },
+  {
+    statueName: "Joseph Edgar Boehm"
+  }
+]
+
+export default SlaverStatues;
```
Comment (benfurber) [inline src/Statues/slavers.js:10]: Missing line break?

File: src/components/StatueList.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      statueName: "Edward Colston",
+    },
+    {
+      statueName: "Marquis de Lafayette",
+    }
+  ]
+  const wrapper = shallow (<StatueList  statue = {MockSlaverStatues} />);
+  expect(wrapper.text()).toContain("Edward Colston")
+  expect(wrapper.text()).toContain("Marquis de Lafayette")
+})
```
Comment (benfurber) [inline src/components/StatueList.spec.js:38]: Is a linter setup and auto-updating in your editor on save?

File: src/components/StatueList.spec.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    {
+      statueName: "Edward Colston",
+    },
+    {
+      statueName: "Marquis de Lafayette",
+    }
+  ]
+  const wrapper = shallow (<StatueList  statue = {MockSlaverStatues} />);
+  expect(wrapper.text()).toContain("Edward Colston")
+  expect(wrapper.text()).toContain("Marquis de Lafayette")
```
Comment (benfurber) [inline src/components/StatueList.spec.js:37]: 👍

File: src/components/StatueListContainer.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React, { Component } from "react";
+import StatueList from "./Statuelist";
+import SlaverStatues from "../Statues/slavers";
+
+export class StatueListContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      statues: SlaverStatues,
```
Comment (benfurber) [inline src/components/StatueListContainer.js:9]: There's no need to put this in state.

File: src/Statues/slavers.js (hunk 4 above)
Comment (benfurber) [reply src/Statues/slavers.js:1]: Also, should be a singular.

Comment (benfurber) [inline src/Statues/slavers.js:1]: I think this should be named `slaverStatueList`.

File: src/Statues/slavers.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+const SlaverStatues = [ 
+  {
+    statueName: "Edward Colston"
```
Comment (benfurber) [inline src/Statues/slavers.js:3]: As the object is named statue, I don't think each element needs it. So `name` is fine.

File: src/components/StatueList.spec.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+import StatueList from "./Statuelist";
+
+it("renders without crashing", () => {
+  const MockSlaverStatues = [
+    {
+      statueName: "Edward Colston",
+    },
+  ];
```
Comment (benfurber) [inline src/components/StatueList.spec.js:10]: Worth either using a proper factory package for this, or at least using this into a factory file.

File: src/components/StatueList.spec.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  shallow(<StatueList statue={MockSlaverStatues} />);
+});
+
+it("renders name of statue", () => {
+  const MockSlaverStatues = [
+    {
+      statueName: "Edward Colston",
+    },
+  ];
+  const wrapper = shallow(<StatueList statue={MockSlaverStatues} />);
```
Comment (benfurber) [inline src/components/StatueList.spec.js:21]: Not sure `statue` is the right prop name. Got any alternatives?

File: src/components/StatueListContainer.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Statues</h1>
```
Comment (benfurber) [inline src/components/StatueListContainer.js:16]: Worth creating a labels file for this.

File: src/components/StatueListContainer.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+        <StatueList statue={this.state.statues} />
+      </div>
+    );
+  }
+}
+
+export default StatueListContainer;
```
Comment (benfurber) [inline src/components/StatueListContainer.js:23]: How else could this (and the other) exports be done?

File: src/components/StatueListContainer.spec.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import StatueListContainer from './StatueListContainer';
+
+it('renders without crashing', () => {
+  shallow(< StatueListContainer />)
```
Comment (benfurber) [inline src/components/StatueListContainer.spec.js:6]: Missing an expectation here?

File: src/components/Statuelist.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    super();
+    this.state = {};
+  }
+
+  render() {
+    const items = this.props.statue.map((item, key) => (
+      <li key={item.statueName}>{item.statueName}</li>
+    ));
+
+    console.log(items);
```
Comment (benfurber) [inline src/components/Statuelist.js:14]: Did you mean to commit this?

File: src/components/Statuelist.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    return (
+      <div>
+        <p>{this.props.statue[0].statueName}</p>
```
Comment (benfurber) [inline src/components/Statuelist.js:17]: This is a WIP right?

File: src/index.js [hunk 17]
```diff
@@ -1,8 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import * as serviceWorker from './serviceWorker';
```
Comment (benfurber) [inline src/index.js:5]: I don't think this is wise. Worth looking into what service workers do.

**Comment by benfurber** [comment]
Worth prefixing your branch names with your initials, a good convention to get in the habit of.

File: src/Statues/slavers.js (hunk 4 above)
Comment (Gina-Frankel) [reply src/Statues/slavers.js:1]: done

File: src/components/StatueList.spec.js (hunk 10 above)
Comment (Gina-Frankel) [reply src/components/StatueList.spec.js:10]: We will do this for the next test

File: src/components/StatueList.spec.js (hunk 11 above)
Comment (Gina-Frankel) [reply src/components/StatueList.spec.js:21]: hmmm we were thinking 
- objectList
- StatueList (but we have used that name for a component) 
-StatueObjectList
-StatueInformation
-StatueData
-SlaverStatueData

File: src/components/StatueList.spec.js (hunk 6 above)
Comment (Gina-Frankel) [reply src/components/StatueList.spec.js:38]: it is set up - we are looking into auto update

File: src/components/Statuelist.js (hunk 16 above)
Comment (Gina-Frankel) [reply src/components/Statuelist.js]: yes that should have been deleted

File: src/components/StatueListContainer.js (hunk 12 above)
Comment (Gina-Frankel) [reply src/components/StatueListContainer.js:10]: We are a bit confused by what you mean exactly - Could you explain it a bit more ...

File: src/components/StatueListContainer.js (hunk 13 above)
Comment (Gina-Frankel) [reply src/components/StatueListContainer.js:17]: We tried to do a named export but it did not seem to work ... 
We will look into it again a bit later

Comment (Gina-Frankel) [reply src/components/StatueListContainer.js:17]: We tried to do a named export but it did not seem to work ... 
We will look into it again a bit later

**Pull request merged**